examples/example_fastapi/main.py

```diff
@@ -60,10 +73,46 @@ async def read_aleph_messages():
     return {"Messages": data}
 
 
+@app.get("/dns")
+async def resolve_dns_hostname():
+    info_inet, info_inet6 = socket.getaddrinfo("example.org", 80, proto=socket.IPPROTO_TCP)
+    ipv4 = info_inet[4][0]
+    ipv6 = info_inet6[4][0]
+    return {
+        "ipv4": ipv4,
+        "ipv6": ipv6,
+    }
+
+
+@app.get("/ip/address")
+async def hello():
+    output = subprocess.check_output(["ip", "addr"], shell=False)
+    return PlainTextResponse(content=output)
+
+
+@app.get("/ip/4")
+async def connect_ipv4():
+    timeout = aiohttp.ClientTimeout(total=5)
+    async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(), timeout=timeout) as session:
+        async with session.get("https://9.9.9.9") as resp:
+            (resp.status == 404) or resp.raise_for_status()
```

Comment:
```suggestion
            if resp.status != 404:
                resp.raise_for_status()
```

Or raise a custom exception with the body of the response for easier debugging.